api/registrations/serializers.py

```diff
@@ -543,6 +543,8 @@ def create(self, validated_data):
                                              ' registered: {}'.format(', '.join(orphan_files_names)))
 
         try:
+            # Still validating metadata, but whether `registration_responses` or `registration_metadata` were populated
+            # on the draft, the other field was built and populated as well.  Both should exist.
             draft.validate_metadata(metadata=draft.registration_metadata, reviewer=reviewer, required_fields=True)
         except ValidationValueError:
             log_exception()  # Probably indicates a bug on our end, so log to sentry
```

Comment:
This area validates a draft's `registration_metadata` before attempting to create a registration, this time, asserting that all the required fields are present.   We turned off throwing an error here when we started using v2 create registrations as part of partial registrations.  Our jsonschema validation wasn't perfect, so we were getting errors that weren't our fault. 


Right now, we don't throw an error, we just log to sentry if there's an error. https://openscience.atlassian.net/browse/ENG-366 should check if we're still getting sentry errors that are all fault, and if we're not, we should restore throwing an error here.